esp/services/esdl_svc_engine/esdl_svc_custom.cpp

```diff
@@ -0,0 +1,346 @@
+    ForEach(*conditionalIterator)
+    {
+        auto xslchooseelement = &conditionalIterator->query();
+        addChildTransformClause(new CEsdlCustomTransformChoose(xslchooseelement), otherwise);
+    }
+}
+
+void CEsdlCustomTransformChoose::compileClauses(IPropertyTreeIterator * rulesiter, bool otherwise, bool typeIsSet)
+{
+    ForEach(*rulesiter)
+    {
+        IPropertyTree & cur = rulesiter->query();
+
+        bool optional = cur.getPropBool("@optional", false);
+        const char * ruleName = cur.queryProp("@name");
+        const char * targetFieldPath = cur.queryProp("@target");
+        if (!targetFieldPath || !*targetFieldPath)
+        {
+            VStringBuffer msg("Encountered custom transform rule '%s' without TargetField path", ruleName ? ruleName : "No Name Provided");
+            if(!optional)
+                throw MakeStringException(-1, "%s", msg.str());
+            else
+                ERRLOG("%s", msg.str());
+                continue;
+        }
+
+        const char * pathToValue = cur.queryProp("@value");
+        if (!pathToValue || !*pathToValue)
+        {
+            VStringBuffer msg("Encountered custom transform rule '%s' without Value path", ruleName ? ruleName : "No Name Provided");
+            if(!optional)
+                throw MakeStringException(-1, "%s", msg.str());
+            else
+                ERRLOG("%s", msg.str());
+                continue;
+        }
+
+        addTransformClause(new CEsdlCustomTransformRule(ruleName, targetFieldPath, pathToValue, optional, typeIsSet ), otherwise);
+    }
+}
+
+void CEsdlCustomTransformChoose::compileClauses(IPropertyTree * rules, bool otherwise)
+{
+    compileClauses(rules->getElements("xsdl:SetValue"), otherwise, true);
+    compileClauses(rules->getElements("xsdl:AppendValue"), otherwise, false);
+}
+
+void CEsdlCustomTransformChoose::addTransformClause(CEsdlCustomTransformRule * fieldmapping, bool otherwise)
+{
+    if (!otherwise)
+        m_chooseClauses.append(*LINK(fieldmapping));
+    else
+        m_otherwiseClauses.append(*LINK(fieldmapping));
+}
+
+void CEsdlCustomTransformChoose::addChildTransformClause(CEsdlCustomTransformChoose * nestedConditional, bool otherwise)
+{
+    if(!otherwise)
+        m_childChooseClauses.append(*LINK(nestedConditional));
+    else
+        m_childOtherwiseClauses.append(*LINK(nestedConditional));
+}
+
+bool CEsdlCustomTransformChoose::evaluate(IXpathContext * xpathContext)
+{
+    bool evalresp = false;
+    try
+    {
+        evalresp = xpathContext->evaluateAsBoolean(m_compiledConditionalXpath.getLink());
+    }
+    catch (...)
+    {
+        DBGLOG("CEsdlCustomTransformChoose:evaluate: Could not evaluate xpath '%s'", xpathContext->getXpath());
+    }
+    return evalresp;
+}
+
+void CEsdlCustomTransformChoose::process(IEspContext * context, IPropertyTree *request, IXpathContext * xpathContext)
+{
+    bool result = false;
+    try
+    {
+        bool result = evaluate(xpathContext);
```

Comment:
What's the purpose of the result variable? It's not being used. Should you change the return value of the method to bool?